crates/vm/levm/src/precompiles.rs

```diff
@@ -887,10 +892,113 @@ pub fn blake2f(
+}
+
+fn verify_kzg_proof(
+    commitment_bytes: &[u8; 48],
+    x: &[u8; 32],
+    y: &[u8; 32],
+    proof_bytes: &[u8; 48],
+) -> Result<bool, VMError> {
+    let commitment_bytes =
+        Bytes48::from_slice(commitment_bytes).map_err(|_| PrecompileError::EvaluationError)?; // Could be ParsingInputError
+    let z_bytes = Bytes32::from_slice(x).map_err(|_| PrecompileError::EvaluationError)?;
+    let y_bytes = Bytes32::from_slice(y).map_err(|_| PrecompileError::EvaluationError)?;
+    let proof_bytes =
+        Bytes48::from_slice(proof_bytes).map_err(|_| PrecompileError::EvaluationError)?;
+
+    let settings =
+        KzgSettings::load_trusted_setup_file().map_err(|_| PrecompileError::EvaluationError)?;
+
+    kzg_rs::kzg_proof::KzgProof::verify_kzg_proof(
+        &commitment_bytes,
+        &z_bytes,
+        &y_bytes,
+        &proof_bytes,
+        &settings,
+    )
+    .map_err(|_| VMError::PrecompileError(PrecompileError::EvaluationError))
+}
+
 fn point_evaluation(
-    _calldata: &Bytes,
-    _gas_for_call: u64,
-    _consumed_gas: &mut u64,
+    calldata: &Bytes,
+    gas_for_call: u64,
+    consumed_gas: &mut u64,
 ) -> Result<Bytes, VMError> {
-    Ok(Bytes::new())
+    if calldata.len() != 192 {
+        return Err(VMError::PrecompileError(PrecompileError::ParsingInputError));
+    }
+
+    // Consume gas
+    let gas_cost = 50000;
+    increase_precompile_consumed_gas(gas_for_call, gas_cost, consumed_gas)?;
+
+    // Parse inputs
+    let hash = calldata
+        .get(..32)
+        .ok_or(PrecompileError::ParsingInputError)?;
+    let mut hash_bytes = [0; 32];
+    hash_bytes.copy_from_slice(hash);
+    let versioned_hash = H256::from(hash_bytes);
+
+    let x = calldata
+        .get(32..64)
+        .ok_or(PrecompileError::ParsingInputError)?;
+    let mut x_bytes = [0; 32];
+    x_bytes.copy_from_slice(x);
+
+    let y = calldata
+        .get(64..96)
+        .ok_or(PrecompileError::ParsingInputError)?;
+    let mut y_bytes = [0; 32];
+    y_bytes.copy_from_slice(y);
+
+    let commitment = calldata
+        .get(96..144)
+        .ok_or(PrecompileError::ParsingInputError)?;
+    let mut commitment_bytes = [0; 48];
+    commitment_bytes.copy_from_slice(commitment);
+
+    let proof = calldata
+        .get(144..192)
+        .ok_or(PrecompileError::ParsingInputError)?;
+    let mut proof_bytes = [0; 48];
+    proof_bytes.copy_from_slice(proof);
+
+    // Perform the evaluation
+
+    // This checks if the commitment is equal to the versioned hash
+    if kzg_commitment_to_versioned_hash(&commitment_bytes) != versioned_hash {
+        return Err(VMError::PrecompileError(PrecompileError::ParsingInputError));
+    }
+
+    // This verifies the proof from a point (x, y) and a commitment
+    if !verify_kzg_proof(&commitment_bytes, &x_bytes, &y_bytes, &proof_bytes).unwrap_or(false) {
+        return Err(VMError::PrecompileError(PrecompileError::ParsingInputError));
+    }
+
+    // Create the output (it's always the same)
```

Comment:
Well, if it is always the same we can avoid calculating it all the time, right? Can a constant replace the code below?

Reply:
Done in [Make output a constant vec of bytes](https://github.com/lambdaclass/ethrex/pull/1569/commits/22d79b5b3fae1261c1d0c3c61bc3b56ec384d3a9). I tried to explain it with comments